### PR TITLE
Client session tutorial

### PR DESCRIPTION
**Pull request by Mats-SX**
~Based on #587~

### CODE REVIEW

**Comment by Mats-SX** [comment]
[[🍓] Client documentation](https://trello.com/c/pCgtIdwz/9125-%F0%9F%8D%93-client-documentation)

File: examples/gds-sessions.ipynb [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+    "<a target=\"_blank\" href=\"https://colab.research.google.com/github/neo4j/graph-data-science-client/blob/main/examples/gds-sessions.ipynb\">\n",
+    "  <img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/>\n",
+    "</a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This Jupyter notebook is hosted [here](https://github.com/neo4j/graph-data-science-client/blob/main/examples/gds-sessions.ipynb) in the Neo4j Graph Data Science Client Github repository.\n",
```
Comment (FlorentinD) [inline examples/gds-sessions.ipynb:27]: you might want to run `/scripts/nb2doc/convert.sh` to create the rendered version for our docs as well

File: examples/gds-sessions.ipynb [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+    "    (annie)-[:KNOWS]->(jeff),\n",
+    "    (annie)-[:KNOWS]->(brie),\n",
+    "    (matt)-[:KNOWS]->(brie),\n",
+    "    (brie)-[:KNOWS]->(elsa),\n",
+    "    (brie)-[:KNOWS]->(jeff),\n",
+    "    (john)-[:KNOWS]->(jeff);\n",
+    "\"\"\"\n",
+    "\n",
+    "# making sure the database is actually empty\n",
+    "assert gds.run_cypher(\"MATCH (n) RETURN count(n)\").squeeze() == 0\n",
```
Comment (FlorentinD) [inline examples/gds-sessions.ipynb:179]: NIT: you can add a custom message to asserts 
```suggestion
    "# making sure the database is actually empty\n",
    "assert gds.run_cypher(\"MATCH (n) RETURN count(n)\").squeeze() == 0, "The database is not empty!"\n",
```

File: examples/gds-sessions.ipynb [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+   "source": [
+    "from graphdatascience.session import GdsPropertyTypes\n",
+    "\n",
+    "G, result = gds.graph.project(\n",
+    "    \"people-and-fruits\",\n",
+    "    \"\"\"\n",
+    "    CALL {\n",
+    "        MATCH (p1:Person)\n",
+    "        OPTIONAL MATCH (p1)-[r:KNOWS]->(p2:Person)\n",
+    "        RETURN p1 AS source, r AS rel, p2 AS target, p1 {.age, .experience, .hipster } AS sourceNodeProperties, p2 {.age, .experience, .hipster } AS targetNodeProperties\n",
```
Comment (FlorentinD) [inline examples/gds-sessions.ipynb:211]: How about splitting this line?

File: examples/gds-sessions.ipynb [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+    "gds.run_cypher(\"MATCH (n) RETURN count(n) AS nodeCount\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Projecting Graphs\n",
+    "\n",
+    "Now that we have imported a graph to our database, we can project it into our GDS Session.\n",
```
Comment (FlorentinD) [inline examples/gds-sessions.ipynb:193]: should we explain the remote projection query a bit?

File: examples/gds-sessions.ipynb [hunk 5]
```diff
@@ -0,0 +1,392 @@
...
+    "        \"experience\": GdsPropertyTypes.LONG,\n",
+    "        \"hipster\": GdsPropertyTypes.LONG,\n",
+    "        \"tropical\": GdsPropertyTypes.LONG,\n",
+    "        \"sourness\": GdsPropertyTypes.DOUBLE,\n",
+    "        \"sweetness\": GdsPropertyTypes.DOUBLE,\n",
+    "    },\n",
+    "    relationshipPropertySchema={},\n",
+    ")\n",
+    "\n",
+    "str(G)"
```
Comment (FlorentinD) [inline examples/gds-sessions.ipynb:236]: if we just write `G` here. The executed cell should show its str version anyway

**Review by FlorentinD — COMMENTED** [review]
Some smaller comments.
Maybe @nvitucci wants to have a look as well?

**Comment by nvitucci** [comment]
> Some smaller comments. Maybe @nvitucci wants to have a look as well?

I just gave a quick look and looks good to me. I'd suggest to remove bits like the "Good luck" at the end and replace some words ("exemplifies", "illustrate") with more common ones, but other than that it would mostly be nit-picking :)

File: examples/gds-sessions.ipynb (hunk 5 above)
Comment (soerenreichardt) [reply examples/gds-sessions.ipynb:236]: No, the formatted string of `G` is very verbose and shows the whole configuration including the query. Wrapping this in `str(G)` produces a nicer output.

File: examples/gds-sessions.ipynb (hunk 4 above)
Comment (Mats-SX) [reply examples/gds-sessions.ipynb:193]: Done.

**Comment by nvitucci** [comment]
> @nvitucci Do you have a suggestion for what word to replace "exemplifies", "illustrate" with?

Sorry, I missed this. In case you're still considering this, I think you can just use "show" for both.

**Pull request merged**